### ISSUE TITLE
Failed to initialize CoreCLR, HRESULT: 0x80131500; Kali Linux Rolling 2016

### ISSUE THREAD

**Issue by ViRb3** [dotnet/coreclr#11417]
.NET Core version: 1.1
Target platform:  Kali Linux Rolling 2016, x64
Project built for platform: debian.8-x64
Error: Failed to initialize CoreCLR, HRESULT: 0x80131500

Also tried with a dummy console app from `dotnet new console`, yet the same problem.

**Comment by janvorli** [comment]
The Kali Linux Rolling is currently based on Debian 9. I think the issue is likely caused by a breaking change that Debian 9 did in the openssl - the soname of the libssl.so was changed to 1.0.2. Can you please check if it is the root of the issue here by running `ls /lib/x86_64-linux-gnu/libssl.so*` and pasting the result here?

**Comment by ThePenetrator** [comment]
```
root@KaliServ:~# uname -a && ls /lib/x86_64-linux-gnu/libssl.so*
Linux KaliServ 4.9.0-kali3-amd64 #1 SMP Debian 4.9.18-1kali1 (2017-04-04) x86_64 GNU/Linux
ls: cannot access '/lib/x86_64-linux-gnu/libssl.so*': No such file or directory
root@KaliServ:~#
```

**Comment by janvorli** [comment]
@ThePenetrator thank you and sorry for the incorrect path to the libssl in my question. So this output confirms that it is really the issue I have mentioned. I have fixed it in the https://github.com/dotnet/corefx/pull/19070. 
@gkhanna79 can we estimate when this change will be in the dotnet core sdk 2.0.0 package that we publish on https://github.com/dotnet/cli?

**Comment by karelz** [comment]
@janvorli does it belong to CoreCLR repo? Is it tracking anything actionable here?
Maybe we should move it into CLI repo?

**Comment by janvorli** [comment]
@ThePenetrator, @gkhanna79, @karelz actually, I have just realized that while the dotnet core sdk that we currently publish would not work on the Debian 9 due to the reason I've mentioned, the issue that @ThePenetrator is seeing actually means that something else is failing too even earlier. So let's investigate a bit more here.

@ThePenetrator, the error code means that there was a managed exception thrown during the coreclr initialization. Could you please run  `strace -f -o log.txt dotnet new console` and then share the log.txt with me via https://gist.github.com/?

**Comment by janvorli** [comment]
@ThePenetrator I've tried to install the dotnet core package 1.1 for Debian from http://dot.net and I can repro your behavior, so I don't need the log anymore. The issue is that the System.Globalization.Native.so references ICU libraries version 52 while the debian 9 has ICU version 57. The upcoming dotnet core 2.0 handles that - it can use any installed ICU with version >= 50.
You can try the latest dotnet 2.0.0 preview package from https://github.com/dotnet/cli - you'll see that it fails due to the problem with libssl I've mentioned:
```
No usable version of the libssl was found
Aborted
```
@gkhanna79 can you estimate when we will update the packages on https://github.com/dotnet/cli to a newer version of stuff from corefx repo that would have the fix for openssl that I've merged in PR https://www.github.com/dotnet/corefx/issues/19070?

**Comment by karelz** [comment]
Here's documentation on dogfooding .NET Core 2.0: https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/dogfooding.md

**Comment by gkhanna79** [comment]
> can you estimate when we will update the packages on https://github.com/dotnet/cli to a newer version of stuff from corefx repo

@janvorli We had significant engineering changes in Core-Setup repo (which CLI consumes) and have not had a good build until yesterday. @eerhardt can comment on when CLI will be able to consume the latest Core-Setup (which should contain the latest Core-FX that should have your fix).

**Comment by eerhardt** [comment]
The latest CLI code contains core-setup's build `2.0.0-preview2-25309-07`, which was built yesterday.  However the CLI master branch's official builds have been failing for a few days and are blocked getting any new builds out.

@livarcocc can estimate when a new .NET Core SDK build will be available.

**Comment by ViRb3** [comment]
Any updates on this issue?

**Comment by crozone** [comment]
Any chance the fix for this will be backported to another 1.1 release? Debian 9 is now considered stable and .NET Core 2.0 is still in preview.

**Comment by karelz** [comment]
2.0 should be out soon enough that it does not make sense to backport anything to earlier releases at this point.
Can you use latest Preview as workaround in the meantime? It is pretty stable stack at this point and we take only few critical bug fixes, so even nightly 2.0 builds should be ok to use.
